### PR TITLE
experiment: refactoring of ir extensions in graph-copy PR

### PR DESCRIPTION
**Pull request by crusso**

### CODE REVIEW

File: src/ir_def/arrange_ir.ml [hunk 1]
```diff
@@ -30,17 +30,18 @@ let rec exp e = match e.it with
...
 
-and system { meta; preupgrade; postupgrade; heartbeat; timer; inspect} = (* TODO: show meta? *)
+and system { meta; preupgrade; postupgrade; heartbeat; timer; inspect; _} = (* TODO: show meta? *)
   "System" $$ [
       "Pre" $$ [exp preupgrade];
       "Post" $$ [exp postupgrade];
       "Heartbeat" $$ [exp heartbeat];
       "Timer" $$ [exp timer];
       "Inspect" $$ [exp inspect];
+      (* TODO: stable stuff *)
```
Comment (luc-blaeser) [inline src/ir_def/arrange_ir.ml:44]: I believe it is okay to skip this here (for the moment)?

File: src/ir_def/check_ir.ml [hunk 2]
```diff
@@ -813,12 +813,13 @@ let rec check_exp env (exp:Ir.exp) : unit =
...
     check_exp env'' timer;
     check_exp env'' inspect;
-    check_exp env'' build_stable_actor;
+    check_exp env'' stable_record;
     typ preupgrade <: T.unit;
     typ postupgrade <: T.unit;
     typ heartbeat <: T.unit;
     typ timer <: T.unit;
     typ inspect <: T.unit;
+    typ stable_record <: stable_type;
```
Comment (luc-blaeser) [inline src/ir_def/check_ir.ml:822]: Nice! (I forgot that check in the previous version.)

File: src/ir_def/arrange_ir.ml (hunk 1 above)
Comment (crusso) [reply src/ir_def/arrange_ir.ml:44]: (but after lunch)

**Review by luc-blaeser — APPROVED** [review]
Very nice! Thanks a lot!

File: src/ir_def/arrange_ir.ml (hunk 1 above)
Comment (luc-blaeser) [reply src/ir_def/arrange_ir.ml]: Enjoy lunch!

**Pull request merged**